google/cloud/spanner/internal/connection_impl.cc

```diff
@@ -265,6 +276,52 @@ StatusOr<ResultSet> ConnectionImpl::ExecuteSql(
   return ResultSet(std::move(*reader));
 }
 
+StatusOr<PartitionedDmlResult> ConnectionImpl::ExecutePartitionedDml(
+    SessionHolder& session, google::spanner::v1::TransactionSelector& s,
+    std::int64_t seqno, ExecutePartitionedDmlParams epdp) {
+  if (session.session_name().empty()) {
+    // Since the session may be sent to other machines, it should not be
+    // returned to the pool when the Transaction is destroyed (release=true).
+    auto session_or = GetSession(/*release=*/true);
```

Comment:
I think this is a bug - PartitionedDml transactions are not partitioned among clients like the other Partition* cases, right?  @scotthart can you confirm?

Reply:
FYI I sent #563 - we can continue the discussion there.